src/@hooks/useWebhookStore.tsx

```diff
@@ -0,0 +1,93 @@
+import { create } from "zustand";
+import { shallow } from "zustand/shallow";
+import toPairs from "lodash/toPairs";
+
+import type { Webhook, WebhookForEditing } from "../@types/cms";
+import type { WebhookData } from "../@types/api";
+
+type WebhookStore = WebhookState & {
+  actions: {
+    setWebhook: ({ name, url }: { name?: string; url?: string }) => void;
+    setWebhookList: (webhook: WebhookData) => void;
+    reset: () => void;
+  };
+};
+
+type WebhookState = {
+  webhookList: WebhookForEditing[];
+  webhook: Webhook;
+  webhookErrorMessage: {
+    name: string;
+    url: string;
+    events?: Set<string>;
+  };
+};
```

Comment:
```suggestion
type WebhookStore = {
  actions: {
    setWebhook: ({ name, url }: { name?: string; url?: string }) => void;
    setWebhookList: (webhook: WebhookData) => void;
    reset: () => void;
  };
  webhookList: WebhookForEditing[];
  webhook: Webhook;
  webhookErrorMessage: {
    name: string;
    url: string;
    events?: Set<string>;
  };
};
```

유니온타입으로 타입들이 흩어져있으면 가독성이 좋지 않아서 한 번에 모아져있는게 좋지 않을까 싶습니다!

Reply:
넵 Omit 이용해서 해볼게요~